AUI-3134 Fixes memory leak with multiple instances of A.DD.Drop by not using it

brunobasto opened the pull request:
Hey Jon,

This makes some performance issues with Form Builder less of a problem. The Forms team is working on adding Drag and Drop functionality to the Form Builder in portal. After each drag interaction, they ask the Form Builder to repaint it self. When it does that, I noticed memory allocation going up quite a lot, making the component slower. I think I managed to narrow down the issue to A.DD.Drop instances not getting cleared by the garbage collector. I couldn't fix it in a more generic way, so I simply removed usages of DD.Drop and implemented the functionality they needed in a simpler way. Memory usage is much better now. Let me know if you have any questions.

Thanks

jonmak08 commented:
Hey @brunobasto,

For the some of the changes like https://github.com/jonmak08/alloy-ui/pull/423/commits/d2b6ac97af3d2724d7b72851cebec03bdcaa0e92#diff-1627211b3193c32da90df660749db545L396 and https://github.com/jonmak08/alloy-ui/pull/423/commits/d2b6ac97af3d2724d7b72851cebec03bdcaa0e92#diff-1627211b3193c32da90df660749db545L637 you are changing the CSS classes that get added to the ```rowNode```.  Should these CSS classes be added to the ```dropNode``` like it was doing before just to the row?

I only ask because this is breaking one of the unit test's that were created for the layout builder (```/aui-layout-builder-resize-col-tests.html```).

![image](https://user-images.githubusercontent.com/860951/32752920-d9e0cbac-c87f-11e7-8aa2-bfa608ff9286.png)

Can you review and update you pull if needed for these changes and/or the unit tests?

Please let me know if you have any questions. Thanks!

brunobasto commented:
Hey Jon, sure. I'll review it. I tried running the unit tests but couldn't manage to get them to run. Will try again.

brunobasto commented:
@jonmak08, I changed some of the tests. They are passing now. I thought it would be more performatic of we just added a class to the row instead of looping through all the nodes and setiting display:none
